servers/src/grin/sync.rs

```diff
@@ -147,6 +148,10 @@ pub fn run_sync(
 							match status {
 								SyncStatus::TxHashsetDownload => (),
 								_ => {
+									// Reset sync_head to header_head
+									// on transition to HeaderSync.
+									chain.init_sync_head(&header_head).unwrap();
```

Comment:
Ignore all the rustfmt changes, this is the important change.

Reply:
👍 
And some review comments:
1. Once entering syncing and `header_sync_due`, it will be called. that means this `init_sync_head()` could be called in every 10 seconds. Do we really need update so frequently?
2. `init_sync_head()` should be called on front of `header_sync()`.
3. If it doesn't matter whether we reset before receiving new Headers or after receiving new Headers, I still prefer after, just let `get_locator()` using `header_head` instead of `sync_head` in case of they're not matched. It make sense to update `sync_head` ONLY after we really received that Headers, doesn't it?
4. If we reset `sync_head` in this way, we will lose the ability to convince ourselves in the future about whether this `sync_head` comes from Sync OR come from this Reset.

@ignopeverell  what's your opinion on this?

Reply:
In another words: 

IMO, the problem of this issue #1524 is in the `get_locator()`, not about the `sync_head` updating / or reset.

Reply:
1. I agree we need this change.
2. I agree we also need to guard against unwanted state changes that could mess this up.
3. Tackle the other issues separately in additional PRs.

Reply:
Please feel free to use @antiochp 's fix solution here 😄   if nobody agree `get_locator()` is a better solution, leaving `sync_head` untouched if no new Headers received.

Anyway, I continue to cleanup the code of #1531, to split the unrelated modification into new PR.